pkg/controllers/workloadrebalancer/workloadrebalancer_controller.go

```diff
@@ -84,18 +92,34 @@ func (c *RebalancerController) Reconcile(ctx context.Context, req controllerrunt
 	}
 
 	// 2. get and update referenced binding to trigger a rescheduling
-	newStatus, successNum, retryNum := c.doWorkloadRebalance(ctx, rebalancer)
+	result := c.doWorkloadRebalance(ctx, rebalancer)
 
-	// 3. update status of WorkloadRebalancer
-	if err := c.updateWorkloadRebalancerStatus(ctx, rebalancer, newStatus); err != nil {
-		return controllerruntime.Result{}, err
+	// 3. update status of WorkloadRebalancer if statusChanged
+	if result.statusChanged {
+		if err := c.updateWorkloadRebalancerStatus(ctx, rebalancer, result.newStatus); err != nil {
+			return controllerruntime.Result{}, err
+		}
+	}
+
+	klog.V(4).Infof("Finish handling WorkloadRebalancer (%s), %d/%d resource success in all, while %d resource need retry",
+		rebalancer.Name, result.successNum, len(result.newStatus.ObservedWorkloads), result.retryNum)
+
+	if result.retryNum > 0 {
+		return controllerruntime.Result{}, fmt.Errorf("%d resource reschedule triggered failed and need retry", result.retryNum)
 	}
-	klog.Infof("Finish handling WorkloadRebalancer (%s), %d/%d resource success in all, while %d resource need retry",
-		rebalancer.Name, successNum, len(newStatus.ObservedWorkloads), retryNum)
 
-	if retryNum > 0 {
-		return controllerruntime.Result{}, fmt.Errorf("%d resource reschedule triggered failed and need retry", retryNum)
+	// 4. when no more resource needs retry, judge whether it needs cleanup.
+	rebalancer.Status = *result.newStatus
+	if rebalancer.Spec.TTLSecondsAfterFinished != nil && rebalancer.Status.FinishTime != nil {
```

Comment:
This judgement `rebalancer.Status.FinishTime != nil` seems will always be true.

Reply:
updated